packages/lib/modules/pool/pool.helpers.ts

```diff
@@ -615,7 +615,7 @@ export function getBoostedGqlTokens(pool: Pool): ApiToken[] {
   const underlyingTokens = pool.poolTokens
     .flatMap(token =>
       shouldUseUnderlyingToken(token, pool)
-        ? [token.underlyingToken as ApiToken]
+        ? [{ ...token, ...token.underlyingToken } as ApiToken]
```

Comment:
Need to merge in attributes from high level token so we can pass on the weight attribute.